src/dos/drive_fat.cpp

```diff
@@ -1085,18 +1120,21 @@ bool fatDrive::FileUnlink(char * name) {
 		DOS_SetError(DOSERR_FILE_NOT_FOUND);
 		return false;
 	}
-/*
-	Technically correct, but maybe an unwanted obstruction, so inactive for now.
 
-	if(fileEntry.attrib & (DOS_ATTR_SYSTEM | DOS_ATTR_HIDDEN)) {
+	/* Technically correct, but maybe an unwanted obstruction, so inactive
+	 * for now. */
+#if 0
+	if (fileEntry.attrib & (DOS_ATTR_SYSTEM | DOS_ATTR_HIDDEN)) {
 		DOS_SetError(DOSERR_FILE_NOT_FOUND);
 		return false;
 	}
-	if(fileEntry.attrib & DOS_ATTR_READ_ONLY) {
+#endif
```

Comment:
This prevent deleting hidden and system DOS files? 
Maybe we could log this delete attempt so the user is aware of what the application or game is trying to do - incase it fails or crashes?

`"FILESYSTEM: Ignoring delete attempt on "IO.SYS", which is a system file"`

`"FILESYSTEM: Ignoring delete attempt on "DOS4GW~3.TMP", which is a hidden file"`

Reply:
To be honest, I'm not even sure if these are correct. Testing with Norton Commander suggests that they're not.